webapp/channels/src/components/activity_log_modal/components/device_icon.tsx

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import type {MessageDescriptor} from 'react-intl';
+import {useIntl} from 'react-intl';
+
+import {isMessageDescriptor} from 'utils/i18n';
+
+type Props = {
+    devicePicture?: string;
+    deviceTitle: MessageDescriptor | string;
+}
+
+export default function DeviceIcon(props: Props) {
+    const intl = useIntl();
+
+    let title;
+    if (isMessageDescriptor(props.deviceTitle)) {
```

Comment:
0/5 on legibility, but to avoid one import, we can check here `typeof props.deviceTitle === 'string'`.

And also 0/5 on legibility, but we could do:
```
const title = typeof props.deviceTitle === 'string' ? props.deviceTitle : intl.formatMessage(props.deviceTitle);
```

Reply:
I like going with the more descriptive name here than using the `typeof` even though they could do the same thing. For the ternary, I ran into some cases where that was causing the line to be a bit long.

I was thinking about making a helper like `formatToString(i18n: I18n, value: string | MessageDescriptor): string` though because I feel like I've written this logic a few times lately. Ideally, I think we'd be consistent about what we pass around, but that seems like it'd be helpful in a bunch of places